### PR TITLE
Adjusted startup sequence of BNO08X sensors to be more flexible regar…

### PR DESCRIPTION
**Pull request by feldhausenryan**
Adjusted startup sequence of BNO08X sensors to be more flexible regarding packet timing/ordering for reset and productid request/response messages. This replaces inconsistent delay()s through the initialization process of the sensor and should result in a more reliable initial connection to the BNO08X sensors.

### CODE REVIEW

**Comment by feldhausenryan** [comment]
Context for this PR is conversation inside ESP32-C3 + BNO08X thread in the dev-forum section on discord.

File: lib/bno080/BNO080.cpp [hunk 1]
```diff
@@ -1038,13 +1068,8 @@ void BNO080::softReset(void)
...
 
-	//Read all incoming data and flush it
-	delay(50);
-	while (receivePacket() == true)
-		; //delay(1);
-	delay(50);
-	while (receivePacket() == true)
-		; //delay(1);
+	waitForCompletedReset();
+	waitForCompletedReset();
```
Comment (Eirenliel) [inline lib/bno080/BNO080.cpp:1072]: Do we need to do it twice after the change? I think it was doing this for both packets to arrive.

**Review by Eirenliel — CHANGES_REQUESTED** [review]

File: lib/bno080/BNO080.cpp (hunk 1 above)
Comment (feldhausenryan) [reply lib/bno080/BNO080.cpp:1072]: The BNO resets twice on reset. So it sends both messages twice for a total of 4 messages. That said I do not want this merged yet without some extra testing from people who have the same hardware combination.

Comment (Eirenliel) [reply lib/bno080/BNO080.cpp:1072]: I see, thank you.

**Comment by TheButlah** [comment]
I will test this tn on my d1 mini, my ferrous_slime v0.1.0 (no i2c pullups), ferrous_slime v0.4.0 (10kohm i2c pullups), and seeed studio xiao esp32c3. I will test in combination with adafruit bno080 and aliexpress bno080.

**Comment by TheButlah** [comment]
* ferrous_slime v0.4.0:  adafruit bno and aliexpres bno both worked, but the old firmware already worked too on that board. I didn't test these super rigourously, maybe only like twice on each board.
* xiao esp32c3 & aliexpress bno: Failed 3/10 times I tried with: [ERR] I2C: Can't find I2C device on provided addresses, scanning for all I2C devices and returning.
* xiao esp32c3 & adafruit bno: failed 2/10 times with the same error

**Comment by TheButlah** [comment]
@feldhausenryan pointed out in discord that the 
```
[ERR] I2C: Can't find I2C device on provided addresses, scanning for all I2C devices and returning.
```
error happens in i2cscan which hasn't been modified in this PR. So the numbers above are probably just random luck.

**Comment by TheButlah** [comment]
The above i2cscan error is fixed when combined with #221. This pr fixes the 
> [FATAL] [BNO080Sensor:0] Can't connect to BNO085 at address 0x4a

So both PRs should be merged, imo.

Please note that I have not really looked too much at the *implementation* of the code in this PR. I merely tested the results.

**Review by TheButlah — APPROVED** [review]

**Comment by TheButlah** [comment]
BTW, @feldhausenryan it would be cool if you submit this to the main adafruit repo as well. If you have time.

**Review by Eirenliel — APPROVED** [review]
I tried to change boot sequence in #190 but you did it properly, gj thanks

**Comment by unlogisch04** [comment]
@TheButlah i belive it is the SparkFun Library
https://github.com/sparkfun/SparkFun_BNO080_Arduino_Library/blob/main/src/SparkFun_BNO080_Arduino_Library.cpp
They still have the https://github.com/sparkfun/SparkFun_BNO080_Arduino_Library/issues/72 issue open. Maybe somewhere there.

**Comment by Eirenliel** [comment]
It brokey... https://github.com/SlimeVR/SlimeVR-Tracker-ESP/issues/264

**Pull request merged**